src/app/functions/server/dashboard.test.ts

```diff
@@ -472,55 +478,106 @@ describe("getExposureReduction", () => {
       dataBreachTotalExposuresNum: 10,
       dataBreachFixedExposuresNum: 10,
       dataBrokerTotalNum: 10,
-      dataBrokerTotalExposuresNum: 10,
-      dataBrokerFixedExposuresNum: 10,
+      dataBrokerTotalExposuresNum: 8,
+      dataBrokerFixedExposuresNum: 8,
       dataBrokerFixedNum: 10,
+      dataBrokerInProgressExposuresNum: 10,
       dataBrokerInProgressNum: 10,
+      dataBrokerManuallyResolvedExposuresNum: 0,
       totalExposures: 10,
       allExposures: testExposure,
-      sanitizedExposures: [],
+      unresolvedExposures: testExposure,
+      inProgressExposures: testExposure,
       fixedExposures: testExposure,
-      fixedSanitizedExposures: [],
+      inProgressFixedExposures: testExposure,
+      unresolvedSanitizedExposures: [],
+      inProgressFixedSanitizedExposures: [],
     };
 
-    const exposureReduction = getExposureReduction(testSummary, []);
-    expect(exposureReduction).toBe(0);
+    const exposureReduction = getExposureReduction(testSummary);
+    expect(exposureReduction).toBe(80);
   });
 });
 
 describe("getDashboardSummary", () => {
+  // sanity checks
+  const noNegativeCounts = (summary: DashboardSummary) => {
+    for (const k in summary.unresolvedExposures) {
+      expect(
+        summary.unresolvedExposures[k as keyof Exposures]
+      ).toBeGreaterThanOrEqual(0);
+    }
+    for (const k in summary.fixedExposures) {
+      expect(
+        summary.fixedExposures[k as keyof Exposures]
+      ).toBeGreaterThanOrEqual(0);
+    }
+    for (const k in summary.inProgressExposures) {
+      expect(
+        summary.inProgressExposures[k as keyof Exposures]
+      ).toBeGreaterThanOrEqual(0);
+    }
+
+    summary.unresolvedSanitizedExposures.forEach((_, v) => {
+      expect(v).toBeGreaterThanOrEqual(0);
+    });
+
+    summary.inProgressFixedSanitizedExposures.forEach((_, v) => {
+      expect(v).toBeGreaterThanOrEqual(0);
+    });
```

Comment:
What does this do?
I thought the second arg of [`.forEach()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)'s callback function was the `index` which is always 0+:
> `index`
> The index of the current element being processed in the array.

For example, this will log `{v: 0}`, `{v: 1}`, `{v: 2}`:
```js
['a', 'b', 'c'].forEach((_, v) => console.log({v}));
```

So when would `v` less than 0?

Reply:
@pdehaan These elements of the array are key value pairs, so the v would be a `number` and the k would be the key. They should look something like
```
[
{ 'physical-addresses': 2 },
{ 'family-members-names': 10 }
]
```

Reply:
OK, I think I'm still confused, but maybe I need to clone this PR and see if I can run+`console.log()` locally to see if the `v` is 0,1 or 2,10.
From your snippet above it does seem like you're using `Array#forEach()` (and not `Map#forEach()` or `Set#forEach()`), so I'm curious how it's destructing key:value pairs here without something like `Object.entries()` to get the actual values. But it's 8:22PM and I'm just eyeballing it in GitHub, so I'm probably overlooking somethin'.

Reply:
I think Peter's right; if every element is indeed an object with a single property with a single value, then you'll still need to iterate over the entries of that object, e.g.:

```suggestion
    summary.unresolvedSanitizedExposures.forEach((unresolvedSanitizedExposure) => {
      Object.values(unresolvedSanitizedExposure).forEach(count => {
        expect(count).toBeGreaterThanOrEqual(0);
      });
    });

    summary.inProgressFixedSanitizedExposures.forEach((inProgressFixedSanitizedExposure) => {
      Object.values(inProgressFixedSanitizedExposure).forEach(count => {
        expect(count).toBeGreaterThanOrEqual(0);
      });
    });
```

What you could alternatively do, is instead of `type SanitizedExposures = Array<Record<string, number>>`, make it an array of arrays with two elements (aka an array of tuples): `type SanitizedExposures = Array<[string, number]>;`. Then, you should be able to destructure it inline:

```suggestion
    summary.unresolvedSanitizedExposures.forEach(([_exposureType, count]) => {
      expect(count).toBeGreaterThanOrEqual(0);
    });

    summary.inProgressFixedSanitizedExposures.forEach(([_exposureType, count]) => {
      expect(count).toBeGreaterThanOrEqual(0);
    });
```

You should be able to verify that the current method doesn't work by modifying the code to actually set a negative value and see that this test doesn't actually fail.

Reply:
Oh right, thanks @Vinnl @pdehaan, confused myself here

Reply:
Or if you don't want the tuple route, I used this `Array#reduce()` mess to build my test case (which uses `Array#concat()` and `Object.values()` magic to build an array of values; conversely I guess we could flatten all the objects into a single object and then call `Object.values()` once):

```js
const assert = require("node:assert/strict");

const tmp = [
  { 'physical-addresses': 10 },
  { 'family-members-names': -2 }
];

const values = tmp.reduce((acc, item) => acc.concat(Object.values(item)), []);

// [10, -2]
console.log(values);

// "AssertionError: -2 should be >= 0"
values.forEach(value => assert(value >= 0, `${value} should be >= 0`));
```